packages/shell-api/src/shell-bson.ts

```diff
@@ -153,36 +153,36 @@ export default function constructShellBson(bson: any): any {
       }
       throw new MongoshInvalidInputError(`${JSON.stringify(input)} is not a valid ISODate`);
     },
-    BinData: function(subtype, b64string): any { // this from 'help misc' in old shell
+    BinData: function(subtype: number, b64string: string): BSON.Binary { // this from 'help misc' in old shell
       assertArgsDefined(subtype, b64string);
       assertArgsType([subtype, b64string], ['number', 'string']);
       const buffer = Buffer.from(b64string, 'base64');
       return new bson.Binary(buffer, subtype);
     },
-    HexData: function(subtype, hexstr): any {
+    HexData: function(subtype: number, hexstr: string): BSON.Binary {
       assertArgsDefined(subtype, hexstr);
       assertArgsType([subtype, hexstr], ['number', 'string']);
       const buffer = Buffer.from(hexstr, 'hex');
       return new bson.Binary(buffer, subtype);
     },
-    UUID: function(hexstr): any {
+    UUID: function(hexstr: string): BSON.Binary {
       assertArgsDefined(hexstr);
       assertArgsType([hexstr], ['string']);
       // Strip any dashes, as they occur in the standard UUID formatting
       // (e.g. 01234567-89ab-cdef-0123-456789abcdef).
       const buffer = Buffer.from(hexstr.replace(/-/g, ''), 'hex');
       return new bson.Binary(buffer, bson.Binary.SUBTYPE_UUID);
     },
-    MD5: function(hexstr): any {
+    MD5: function(hexstr: string): BSON.Binary {
       assertArgsDefined(hexstr);
       assertArgsType([hexstr], ['string']);
       const buffer = Buffer.from(hexstr, 'hex');
       return new bson.Binary(buffer, bson.Binary.SUBTYPE_MD5);
     }
   };
   [ 'ObjectId', 'Code', 'DBRef', 'MaxKey', 'MinKey', 'Timestamp', 'Symbol', 'Map'].forEach((className) => {
-    (bsonPkg[className] as any).help = (): Help => (helps[className]);
-    Object.setPrototypeOf((bsonPkg[className] as any).help, helps[className]);
+    ((bsonPkg as any)[className] as any).help = (): Help => (helps[className]);
+    Object.setPrototypeOf(((bsonPkg as any)[className] as any).help, helps[className]);
```

Comment:
Maybe we can reduce the amount of `any`s here also with the help of `keyof`:

```ts
// (GitHub doesn't allow me to do a suggestion for all touched lines so just posting a snippet, sorry!)

const keys: (keyof typeof bsonPkg)[] = [
  "ObjectId",
  "Code",
  "DBRef",
  "MaxKey",
  "MinKey",
  "Timestamp",
  "Symbol",
  "Map",
];

keys.forEach((className) => {
  bsonPkg[className].help = (): Help => helps[className];
  Object.setPrototypeOf(bsonPkg[className].help, helps[className]);
});

```

What do you think?

Reply:
And adopted this as well :slightly_smiling_face: